lib/opentelemetry/trace/exporter/otlp.lua

```diff
@@ -7,13 +10,37 @@ local mt = {
     __index = _M
 }
 
-function _M.new(http_client)
+function _M.new(http_client, timeout_ms)
     local self = {
         client = http_client,
+        timeout_ms = timeout_ms or DEFAULT_TIMEOUT_MS,
     }
     return setmetatable(self, mt)
 end
 
+local function call_collector(self, pb_encoded_body)
+    local start_time_ms = util.gettimeofday_ms()
+    local failures = 0
+
+    while failures < RETRY_LIMIT do
+        if util.gettimeofday_ms() - start_time_ms > self.timeout_ms then
+            ngx.log(ngx.WARN, "Collector retries timed out (timeout " .. self.timeout_ms .. ")")
+            break
+        end
+
+        local res, _ = self.client:do_request(pb_encoded_body)
+        if not res then
+            failures = failures + 1
+            if not _TEST then
```

Comment:
Not sure if there's a better way to do this....

Reply:
I prefer to decr RETRY_LIMIT to 3, then remove check `_TEST`.

Reply:
046bc1a564f048d4b98e26ec69df20a25c878552